src/Microsoft.Identity.Client/Azure/Exponential.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Azure
+{
+    /// <summary>
+    /// RetryWithExponentialBackoff runs a task with an exponential backoff
+    /// </summary>
+    public sealed class RetryWithExponentialBackoff
```

Comment:
Maybe integrate existing Http implementation and backoff or look to improve.

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/511a6866a394ea7b252a98078d215985fcd93987/src/Microsoft.Identity.Client/Http/HttpManager.cs

Reply:
I had thought about that, but I was afraid to be too invasive. What I would need to add a retry strategy for the HTTP client implementation as well as the ability to inject a shortened timeout. I'm happy to make those changes, but like I said, I didn't want to add complexity to the existing HTTP client implementation. Thoughts?

Reply:
50/50  maybe the plan was to route everything through that layer. Its nicely coupled inside `ServicePrincipalTokenProvider`

Reply:
Would it make sense to provide it even more generic as an `HttpMessageHandler`?

Reply:
This has been moved to be an internal implementation detail.